File: Python/1. Two Sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mmap = {}

        for indices, value in enumerate(nums):
            if target - value in mmap:
                return [mmap[target - value], indices]
            else:
                mmap[value] = indices

        return

File: 算法/C++/1. Two Sum.cpp

class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {

        vector<int> res;
        unordered_map<int, int> mmap;
        for (int i = 0; i < nums.size(); i++) {
            int left = target - nums[i];
            if (mmap.count(left) > 0) {
                res.push_back(mmap[left]);
                res.push_back(i);
                return res;
            }
            mmap[nums[i]] = i;
        }

        return res;
        
    }
};

//map.count() faster than map.find()